### PR TITLE
Support universal function call syntax in function calls

### LINKED ISSUE #596: Got expected expression in valid function call
The [following](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b1ae23cc8a80a49157ef2343c3b5aaeb) (valid) code results in errors in the [syntax tree](https://gist.github.com/regiontog/458888f109d188e983ae0964cf3c1a30#file-gistfile1-txt-L102).

```rust
struct Foo;

impl Foo {
    fn bar() -> bool {
        unimplemented!()
    }
}

fn baz(_: bool) {}

fn main() {
    baz(<Foo>::bar())
}
```

### PR DESCRIPTION
**Pull request by DJMcNab**
Fix #596

### CODE REVIEW

File: crates/ra_syntax/src/grammar/expressions/atom.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ pub(super) const ATOM_EXPR_FIRST: TokenSet = LITERAL_FIRST.union(token_set![
     L_PAREN,
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/expressions/atom.rs]: Hm, I think we could `LITEREAL_FIRST.unition(PATH_FIRST).uniont` ...

File: crates/ra_syntax/src/grammar/expressions/atom.rs [hunk 2]
```diff
@@ -36,7 +37,7 @@ pub(crate) fn literal(p: &mut Parser) -> Option<CompletedMarker> {
 }
 
 // E.g. for after the break in `if break {}`, this should not match
-pub(super) const ATOM_EXPR_FIRST: TokenSet = LITERAL_FIRST.union(token_set![
+pub(super) const ATOM_EXPR_FIRST: TokenSet = LITERAL_FIRST.union(PATH_FIRST).union(token_set![
```
Comment (matklad) [inline crates/ra_syntax/src/grammar/expressions/atom.rs]: The grammar is setup in such way that no imports are necessary. 

```suggestion
pub(super) const ATOM_EXPR_FIRST: TokenSet = LITERAL_FIRST.union(paths::PATH_FIRST).union(token_set![
```

File: crates/ra_syntax/src/grammar/expressions/atom.rs [hunk 3]
```diff
@@ -1,4 +1,5 @@
 use super::*;
+use crate::grammar::paths::PATH_FIRST;
```
Comment (DJMcNab) [inline crates/ra_syntax/src/grammar/expressions/atom.rs]:
```suggestion
```

This is a test to see how well github's suggested changes work.

**Comment by matklad** [comment]
r=me

**Comment by DJMcNab** [comment]
bors r=matklad

**Pull request merged**